docs/content/contribute/development.md

```diff
@@ -0,0 +1,191 @@
+
+## Postgresql
+
+Download PostgreSQL from https://www.postgresql.org/download/, version >= 9.4.
+
+You can easily use only PostgreSQL binaries, downloaded from https://www.enterprisedb.com/download-postgresql-binaries.
+
+Initialize DB by running:
+
+```bash
+$ mkdir -p ~/data/postgres/data
+$ initdb -D ~/data/postgres/data
+```
+
+Create user `cds` and database `cds`
+
+```
+$ psql -d postgres
+postgres=# create user cds with password 'cds';
+postgres=# create database cds owner cds;
+postgres=\q
+```
+
+Then launch PostgreSQL with:
+
+```bash
+$ postgres -D ~/data/postgres/data
+```
+
+That's all for a local PostgreSQL installation - not recommanded for a production installation.
+
+## Redis
+
+Download the latest stable Redis from https://redis.io/download, version >= 3.2
+
+Example with version 4.0.11:
+
+```bash
+$ wget http://download.redis.io/releases/redis-4.0.11.tar.gz
+$ tar xzf redis-4.0.11.tar.gz
+$ cd redis-4.0.11
+$ make
+# launch redis-server
+$ src/redis-server
+# you should add src/ to your PATH
+```
+
+That's all for a local Redis installation.
+
+
+## NodeJS
+
+Download the latest stable Node from https://nodejs.org/en/download/current/, version >= 10.8.0
+
+Example with version 10.8.0 on OSX:
+
+```bash
+$ curl -O https://nodejs.org/dist/v10.8.0/node-v10.8.0-darwin-x64.tar.gz
+$ tar xzf node-v10.8.0-darwin-x64.tar.gz
+# directory node-v10.8.0-darwin-x64 is created
+# You should add node-v10.8.0-darwin-x64/bin to your PATH
+```
+
+## Golang
+
+Download the latest Golang version from https://golang.org/dl/, version >= 1.10
+
+Example with version 1.10.3 on OSX:
+
+```bash
+$ export GOROOT=~/go
+$ export PATH=$PATH:$GOROOT/bin
+$ cd ~
+$ curl -O https://dl.google.com/go/go1.10.3.darwin-amd64.tar.gz
+$ tar xzf go1.10.3.darwin-amd64.tar.gz
+```
+
+Check if go installation is ok
+
+```bash
+$ go version
+go version go1.10.3 darwin/amd64
+```
+
+## CDS
+
+Compile CDS:
+
+```bash
+# Checkout code
+$ mkdir -p $(go env GOPATH)/src/github.com/ovh
+$ cd $(go env GOPATH)/src/github.com/ovh
+$ git clone https://github.com/ovh/cds.git
+
+# Compile engine
+$ cd $(go env GOPATH)/src/github.com/ovh/cds/engine
+$ go install
+
+# Compile worker
+$ cd $(go env GOPATH)/src/github.com/ovh/cds/engine/worker
+$ go install
+
+# Compile cdsctl
+$ cd $(go env GOPATH)/src/github.com/ovh/cds/cli/cdsctl
+$ go install
+
+# Compile ui
+$ cd $(go env GOPATH)/src/github.com/ovh/cds/ui
+$ npm install
+```
+
+Configure CDS:
+
+```
+# Generate default configuration file
+$ engine config new > ~/.cds/dev.toml
+
+# edit ~/.cds/dev.toml file 
+## in section [api]
+### --> set variable defaultOS to your OS, darwin if you are on osx for example
+## in section [hatchery.local.commonConfiguration]
+### --> set name to "hatchery-local"
+### --> uncomment url, should be set to url = "http://localhost:8086" 
+
+```
+
+Prepare database:
+
+```bash
+$ cd $(go env GOPATH)/src/github.com/ovh/cds/engine
+$ engine database upgrade --db-password cds --db-sslmode disable
```

Comment:
I would like to understand what is the purpose of this command before typing it.